backend/server/mailing.go

```diff
@@ -2,91 +2,239 @@ package main
 			return c.JSON(http.StatusBadRequest, H{
 				"error": err,
 			})
 		}
 
-		// TODO: Compose and send email
-		return c.JSON(http.StatusOK, H{})
+		// Add to bundle
+		switch targetEmail {
+		case sponsorshipEmail:
+			sponsorshipBundle = append(sponsorshipBundle, enquiry)
+		case infoEmail:
+			infoBundle = append(infoBundle, enquiry)
+		}
+
+		return c.JSON(http.StatusAccepted, H{})
 	}
 }
 
-// FeedbackEnquiry - validates feedback input and forwards as an email to the revelant inbox
-func FeedbackEnquiry(targetEmail string) echo.HandlerFunc {
+// HandleFeedback - validates feedback input and forwards as an email to the revelant inbox
+func HandleFeedback() echo.HandlerFunc {
 	return func(c echo.Context) error {
-		name := c.FormValue("name")
-		email := c.FormValue("email")
-		body := c.FormValue("body")
-
-		// if there is no email, use the struct where emails are not required (FeedbackNoEmail)
-		if email == "" {
-			feedback := FeedbackNoEmail{
-				Name:  name,
-				Email: email,
-				Body:  body,
-			}
+		// Extract fields from form
+		feedback := Feedback{
+			Name:  c.FormValue("name"),
+			Email: c.FormValue("email"),
+			Body:  c.FormValue("body"),
+		}
 
-			if err := c.Validate(feedback); err != nil {
-				return c.JSON(http.StatusBadRequest, H{
-					"error": err,
-				})
-			}
+		// Validate struct
+		if err := c.Validate(feedback); err != nil {
+			return c.JSON(http.StatusBadRequest, H{
+				"error": err,
+			})
+		}
 
-			// if there is an email, check that the email is valid by using the Feedback Struct
-		} else {
-			feedback := Feedback{
-				Name:  c.FormValue("name"),
-				Email: c.FormValue("email"),
-				Body:  c.FormValue("body"),
-			}
+		// Add to bundle
+		feedbackBundle = append(feedbackBundle, feedback)
 
-			if err := c.Validate(feedback); err != nil {
-				return c.JSON(http.StatusBadRequest, H{
-					"error": err,
-				})
-			}
+		return c.JSON(http.StatusAccepted, H{})
+	}
+}
 
-		}
+/////////
+// TIMERS
+/////////
+
+// This function is executed once in a subroutine and triggers every 15 minutes
+func enquiryMailingTimer() {
+	for {
+		time.Sleep(15 * time.Minute)
+		DispatchEnquiryBundles()
+	}
+}
+
+// This function is executed once in a subroutine and triggers once every day
+func feedbackMailingTimer() {
+	for {
+		time.Sleep(24 * time.Hour)
+		DispatchFeedbackBundle()
+	}
+}
 
-		//TODO: Compose email and send
-		return c.JSON(http.StatusOK, H{})
+//////////////
+// DISPATCHERS
+//////////////
+
+// DispatchEnquiryBundles - public trigger for dispatching enquiries
+func DispatchEnquiryBundles() {
+	go sendEnquiryBundle(infoEmail, &infoBundle)
+	go sendEnquiryBundle(sponsorshipEmail, &sponsorshipBundle)
+}
+
+// DispatchFeedbackBundle - public trigger for dispatching feedbacks
+func DispatchFeedbackBundle() {
+	go sendFeedbackBundle(infoEmail, &feedbackBundle)
+}
+
+/////////////////
+// BUNDLE SENDERS
+/////////////////
+
+func sendEnquiryBundle(targetEmail string, bundle *[]Enquiry) {
```

Comment:
Hmm can we somehow re-use this function with **sendFeedbackEnquiry**? They seem to be doing similar actions